datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -428,6 +429,13 @@ public SdxCluster getByNameInAccount(String userCrn, String name) {
 
         newSdxCluster.setDatabaseAvailabilityType(sdxCluster.getDatabaseAvailabilityType());
         StackV4Request stackRequest = getStackRequest(shape, sdxCluster.isRangerRazEnabled(), null, cloudPlatform, sdxCluster.getRuntime(), null);
+        if (shape == SdxClusterShape.MEDIUM_DUTY_HA) {
+            // This is added to make sure the host name used by Light and Medium duty are not the same.
+            CustomDomainSettingsV4Request customDomainSettingsV4Request = new CustomDomainSettingsV4Request();
+            customDomainSettingsV4Request.setHostname(sdxCluster.getClusterName() + SDX_RESIZE_NAME_SUFFIX);
```

Comment:
maybe it won't cause any problem, but did you try it with attached DHs? I am not aware of if DHs joining with FQDN to the SDX master but if they do it then it can cause issue.

Reply:
With attached DHs, remote data context has to be updated regardless. It should take care of updating the new configuration with new host names.